src/plugins/intel_gpu/src/runtime/debug_configuration.cpp

```diff
@@ -357,6 +384,23 @@ const debug_configuration *debug_configuration::get_instance() {
 #endif
 }
 
+bool debug_configuration::is_target_dump_prof_data_iteration(int64_t iteration) const {
+#ifdef GPU_DEBUG_CONFIG
+    if (iteration < 0)
+        return true;
+
+    if (!dump_prof_data_iter_params.is_enabled)
```

Comment:
how about checking this before calling this function?

Reply:
Removed this checking so that it can be checked on the side actually called